docs/articles/contributing/architecture.md

```diff
@@ -1,5 +1,40 @@
+> Naiad is a distributed system for executing data parallel, cyclic dataflow programs. It offers the high throughput of batch processors, the low latency of stream processors, and the ability to perform iterative and incremental computations. Although existing systems offer some of these features, applications that require all three have relied on multiple platforms, at the expense of efficiency, maintainability, and simplicity. Naiad resolves the complexities of combining these features in one framework.
+> A new computational model, timely dataflow, underlies Naiad and captures opportunities for parallelism across a wide class of algorithms. This model enriches dataflow computation with timestamps that represent logical points in the computation and provide the basis for an efficient, lightweight coordination mechanism.
+> We show that many powerful high-level programming models can be built on Naiad's low-level primitives, enabling such diverse tasks as streaming data analysis, iterative machine learning, and interactive graph mining. Naiad outperforms specialized systems in their target application domains, and its unique features enable the development of new high-performance applications.
+
+At some point, Microsoft Research changed direction with respect to the Silicon Valley Lab (I don't know the real story, but some of the prior reporting on it is [here](https://www.vox.com/2014/9/18/11631044/microsoft-shuts-down-silicon-valley-research-lab-amid-broader-layoffs)) and in some way or another, the team moved on from there to start other things. Many of the original team kept the idea of dataflow programs core to what they next worked on. Frank went on to spend time working on a Rust version of the core tenants of Naiad and Derek went on to bring the core dataflow programming concepts to tensorflow.
+
+## of Pythons and Crabs
+
+Bytewax leverages another open-source project to bring the Python Native capabilities on top of a Rust processing engine. This project is called PyO3 and is used by some of our friends at [PyDantic], [Polars], [ReRun] and more. PyO3 is a great project that allows you to run Rust functions from Python like a foreign function interface, and it also lets you embed Python in Rust. Many more libraries are starting to leverage this combination of Rust and Python as it provides a very powerful combination of easy to use and broad applicability of Python with the performance and security benefits of Rust. In Bytewax we both call Rust from Python, but also embed Python in Rust. This can get a little tricky and is the less commonly used pattern used in PyO3.
```

Comment:
1. reminder to add links
2. I would add a "calming" sentence towards the end that you don't need to know rust to use bytewax, as the API is pythonic